commit d41da87f988afb2aac35d32df34ecbe04cccae01
Author: Eric Anholt <eric@anholt.net>
Date:   Tue Jan 30 14:21:02 2018 -0800

    drm/vc4: Add missing enable/disable vblank handlers in fkms.
    
    Fixes hang at boot in 4.14.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/gpu/drm/vc4/vc4_crtc.c b/drivers/gpu/drm/vc4/vc4_crtc.c
index a98847773cc8..575681d9cb96 100644
--- a/drivers/gpu/drm/vc4/vc4_crtc.c
+++ b/drivers/gpu/drm/vc4/vc4_crtc.c
@@ -710,15 +710,8 @@ static void vc4_crtc_atomic_flush(struct drm_crtc *crtc,
 
 static int vc4_enable_vblank(struct drm_crtc *crtc)
 {
-	struct drm_device *dev = crtc->dev;
-	struct vc4_dev *vc4 = to_vc4_dev(dev);
 	struct vc4_crtc *vc4_crtc = to_vc4_crtc(crtc);
 
-	if (vc4->firmware_kms) {
-		/* XXX: Can we mask the SMI interrupt? */
-		return 0;
-	}
-
 	CRTC_WRITE(PV_INTEN, PV_INT_VFP_START);
 
 	return 0;
@@ -726,15 +719,8 @@ static int vc4_enable_vblank(struct drm_crtc *crtc)
 
 static void vc4_disable_vblank(struct drm_crtc *crtc)
 {
-	struct drm_device *dev = crtc->dev;
-	struct vc4_dev *vc4 = to_vc4_dev(dev);
 	struct vc4_crtc *vc4_crtc = to_vc4_crtc(crtc);
 
-	if (vc4->firmware_kms) {
-		/* XXX: Can we mask the SMI interrupt? */
-		return;
-	}
-
 	CRTC_WRITE(PV_INTEN, 0);
 }
 
diff --git a/drivers/gpu/drm/vc4/vc4_firmware_kms.c b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
index d5066f395b62..e3dfc20c404c 100644
--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -433,6 +433,19 @@ static int vc4_page_flip(struct drm_crtc *crtc,
 	return drm_atomic_helper_page_flip(crtc, fb, event, flags, ctx);
 }
 
+static int vc4_fkms_enable_vblank(struct drm_crtc *crtc)
+{
+	/* XXX: Need a way to enable/disable the interrupt, to avoid
+	 * DRM warnings at boot time.
+	 */
+
+	return 0;
+}
+
+static void vc4_fkms_disable_vblank(struct drm_crtc *crtc)
+{
+}
+
 static const struct drm_crtc_funcs vc4_crtc_funcs = {
 	.set_config = drm_atomic_helper_set_config,
 	.destroy = drm_crtc_cleanup,
@@ -443,6 +456,8 @@ static const struct drm_crtc_funcs vc4_crtc_funcs = {
 	.reset = drm_atomic_helper_crtc_reset,
 	.atomic_duplicate_state = drm_atomic_helper_crtc_duplicate_state,
 	.atomic_destroy_state = drm_atomic_helper_crtc_destroy_state,
+	.enable_vblank = vc4_fkms_enable_vblank,
+	.disable_vblank = vc4_fkms_disable_vblank,
 };
 
 static const struct drm_crtc_helper_funcs vc4_crtc_helper_funcs = {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1127_d41da87f988afb2aac35d32df34ecbe04cccae01.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1127_d41da87f988afb2aac35d32df34ecbe04cccae01.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
